src/classes/UTIL_Currency.cls

```diff
@@ -77,15 +77,18 @@ public with sharing class UTIL_Currency implements UTIL_Currency.Interface_x {
     /**
      * @description Returns true if Advanced Currency Management is enabled in the org
      * - technically this will return true even if the Admin disables ACM after it's enabled the first time
-     * - TODO: working on a solution for this still
+     * - This is as accurate an assessment as we can get within Apex at present.
      */
     private Boolean isACMEnabled = null;
     public Boolean isAdvancedCurrencyManagementEnabled() {
         if (isACMEnabled == null) {
+            isACMEnabled = false;
             if (isMultiCurrencyOrganization()) {
-                isACMEnabled = Schema.getGlobalDescribe().containsKey('DatedConversionRate');
-            } else {
-                isACMEnabled = false;
+                // Using a string in Database.query because this query won't compile if multicurrency isn't enabled
+                Integer DCRCount = Database.query('SELECT Id FROM DatedConversionRate LIMIT 1').size();
```

Comment:
Since we're technically working around a core bug, I recommend wrapping this query in a try/catch in case they eventually fix the problem of this SObject existing when ACM hasn't actually been enabled.